app/scripts/models/account.js

```diff
@@ -415,6 +419,43 @@ define(function (require, exports, module) {
         .then(function (updatedSessionData) {
           self.set(updatedSessionData);
         });
+    },
+
+    /**
+     * Fetch the account's device list and populate the `devices` collection.
+     *
+     * @param {object} devices - Devices collection
+     * @returns {promise} - resolves when complete
+     */
+    fetchDevices: function (devices) {
+      var sessionToken = this.get('sessionToken');
+
+      return this._fxaClient.deviceList(sessionToken)
+        .then(function (deviceList) {
+          devices.set(deviceList.map(function (device) {
+            device.isCurrent = device.sessionToken === sessionToken;
```

Comment:
@philbooth - if you stop returning the `sessionToken` and instead return `isDeviceCurrent`, this line can be removed.